x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/GetEngineAction.java

```diff
@@ -0,0 +1,146 @@
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.entsearch.engine.Engine;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+import static org.elasticsearch.xpack.entsearch.engine.EngineIndexService.ENGINE_ALIAS_NAME;
+
+public class GetEngineAction extends ActionType<GetEngineAction.Response> {
+
+    public static final GetEngineAction INSTANCE = new GetEngineAction();
+    public static final String NAME = "indices:admin/engine/get";
+
+    private GetEngineAction() {
+        super(NAME, GetEngineAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest implements IndicesRequest.Replaceable {
+
+        public static final IndicesOptions DEFAULT_INDICES_OPTIONS = IndicesOptions.strictExpandOpen();
+
+        private String[] names = new String[] { ENGINE_ALIAS_NAME };
+        private final IndicesOptions indicesOptions = DEFAULT_INDICES_OPTIONS;
+
+        private final String engineId;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.engineId = in.readString();
+        }
+
+        public Request(String engineId) {
+            this.engineId = engineId;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            if (engineId == null || engineId.isEmpty()) {
+                validationException = addValidationError("engineId missing", validationException);
+            }
+
+            return validationException;
+        }
+
+        public String getEngineId() {
+            return engineId;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(engineId);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request request = (Request) o;
+            return Objects.equals(engineId, request.engineId);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(engineId);
+        }
+
+        @Override
+        public IndicesRequest indices(String... indices) {
+            this.names = indices;
+            return this;
+        }
+
+        @Override
+        public String[] indices() {
+            return names;
```

Comment:
As Jim was mentioning - this should return the Engine alias name. We probably want that method to be part of the Engine class so it's easier to access from other parts of the code.

Reply:
Fixed.